### PR TITLE
fix(eslint-plugin): [use-component-selector] applies to template literals

### LINKED ISSUE #800: [@angular-eslint/use-component-selector] Component Selector does not support Backticks
**Description and reproduction of the issue**

When the selector name is in doublequotes ("app-root") it does not throw any error.

But when selector name is in Backticks (`app-root`) it throws an error -> "The selector of the component is mandatoryeslint@angular-eslint/use-component-selector"

image

```JSON
{
  "rules": {
    "@angular-eslint/<rule>": "@angular-eslint/use-component-selector"
  }
}
```

```TS
@Component ( {
  selector    : `app-root`,
  templateUrl : `./app.component.html`,
  styleUrls   : [ `./app.component.scss` ],
} )
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    |"13.0.0-alpha.0"|
| `@typescript-eslint/parser`       |"^5.3.1"|
| `ESLint`                           |"^8.2.0"|
| `node`                           |"v16.13.0"|

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 13.0.2
Node: 16.13.0
Package Manager: npm 8.1.3
OS: linux x64

Angular: 13.0.1
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1300.2
@angular-devkit/build-angular   13.0.2
@angular-devkit/core            13.0.2
@angular-devkit/schematics      13.0.2
@angular/cli                    13.0.2
@schematics/angular             13.0.2
rxjs                            7.4.0
typescript                      4.4.4

```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

### PR DESCRIPTION
**Pull request by hyperupcall**
Closes #800

### CODE REVIEW

File: packages/eslint-plugin/src/rules/use-component-selector.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export default createESLintRule<Options, MessageIds>({
           return;
         }
 
+        if (selector && ASTUtils.isTemplateLiteral(selector)) {
```
Comment (abaran30) [inline packages/eslint-plugin/src/rules/use-component-selector.ts:36]: Could this check just be added to the original if statement? Also, would you be able to add a test case or two to ensure this works?

**Review by abaran30 — CHANGES_REQUESTED** [review]

**Comment by hyperupcall** [comment]
@abaran30  Sorry for the wait - combined the `if` statements and created additional tests to match.

<details>

<summary>Test results</summary>

```txt
$ ./node_modules/.bin/nx run eslint-plugin:test --skip-nx-cache

 >  NX   --skip-nx-cache disables the connection to Nx Cloud for the current run.

   The remote cache will not be read from or written to during this run.


   ✔    2/2 dependent project tasks succeeded [0 read from cache]

   Hint: you can run the command with --verbose to see the full dependent project outputs

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————


> nx run eslint-plugin:build


> nx run eslint-plugin:test

● Validation Warning:

  Unknown option "coverageDirectory" with value "../../coverage/packages/eslint-plugin" was found.
  This is probably a typing mistake. Fixing it will remove this message.

  Configuration Documentation:
  https://jestjs.io/docs/configuration

● Validation Warning:

  Unknown option "coverageReporters" with value ["text-summary", "lcov"] was found.
  This is probably a typing mistake. Fixing it will remove this message.

  Configuration Documentation:
  https://jestjs.io/docs/configuration

ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
ts-jest[ts-jest-transformer] (WARN) Define `ts-jest` config under `globals` is deprecated. Please do
transform: {
    <transform_regex>: ['ts-jest', { /* ts-jest config goes here in Jest */ }],
},
ts-jest[ts-jest-transformer] (WARN) Got a `.js` file to compile while `allowJs` option is not set to `true` (file: /storage/ur/storage_home/Docs/Programming/Repositories/fox-forks/angular-eslint/packages/utils/dist/index.js). To fix this:
  - if you want TypeScript to process JS files, set `allowJs` to `true` in your TypeScript config (usually tsconfig.json)
  - if you do not want TypeScript to process your `.js` files, in your Jest config change the `transform` key which value is `ts-jest` so that it does not match `.js` files anymore
ts-jest[ts-jest-transformer] (WARN) Got a `.js` file to compile while `allowJs` option is not set to `true` (file: /storage/ur/storage_home/Docs/Programming/Repositories/fox-forks/angular-eslint/packages/utils/dist/eslint-plugin/ast-utils.js). To fix this:
  - if you want TypeScript to process JS files, set `allowJs` to `true` in your TypeScript config (usually tsconfig.json)
  - if you do not want TypeScript to process your `.js` files, in your Jest config change the `transform` key which value is `ts-jest` so that it does not match `.js` files anymore
ts-jest[ts-jest-transformer] (WARN) Got a `.js` file to compile while `allowJs` option is not set to `true` (file: /storage/ur/storage_home/Docs/Programming/Repositories/fox-forks/angular-eslint/packages/utils/dist/eslint-plugin/rule-fixes.js). To fix this:
  - if you want TypeScript to process JS files, set `allowJs` to `true` in your TypeScript config (usually tsconfig.json)
  - if you do not want TypeScript to process your `.js` files, in your Jest config change the `transform` key which value is `ts-jest` so that it does not match `.js` files anymore
ts-jest[ts-jest-transformer] (WARN) Got a `.js` file to compile while `allowJs` option is not set to `true` (file: /storage/ur/storage_home/Docs/Programming/Repositories/fox-forks/angular-eslint/packages/utils/dist/eslint-plugin/selector-utils.js). To fix this:
  - if you want TypeScript to process JS files, set `allowJs` to `true` in your TypeScript config (usually tsconfig.json)
  - if you do not want TypeScript to process your `.js` files, in your Jest config change the `transform` key which value is `ts-jest` so that it does not match `.js` files anymore
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/prefer-on-push-component-change-detection/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/component-max-inline-declarations/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/use-component-selector/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/configs.test.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-output-native/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/contextual-decorator/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/contextual-lifecycle/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/sort-ngmodule-metadata-arrays/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/use-injectable-provided-in/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-output-rename/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/component-selector/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-input-rename/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-output-on-prefix/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/require-localize-metadata/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/directive-class-suffix/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/component-class-suffix/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-input-prefix/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-pipe-impure/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-empty-lifecycle-method/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-host-metadata-property/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-attribute-decorator/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/prefer-standalone-component/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/directive-selector/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/use-pipe-transform-interface/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-inputs-metadata-property/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-outputs-metadata-property/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-lifecycle-call/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-conflicting-lifecycle/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/relative-url-prefix/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/use-lifecycle-interface/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/use-component-view-encapsulation/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-queries-metadata-property/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/no-forward-ref/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/prefer-output-readonly/spec.ts
 PASS   eslint-plugin  packages/eslint-plugin/tests/rules/pipe-prefix/spec.ts

Test Suites: 35 passed, 35 total
Tests:       626 passed, 626 total
Snapshots:   0 total
Time:        13.062 s
Ran all test suites.
```

</details>

**Review by JamesHenry — CHANGES_REQUESTED** [review]
Thanks @hyperupcall!

Our docs are generated from our rule implementations and their unit tests, so you please need to regenerate them and commit the results.

```sh
yarn update-rule-docs
```

**Comment by hyperupcall** [comment]
Oops! Updated the docs :+1:

**Comment by JamesHenry** [comment]
Thanks!

**Pull request merged**